src/controllers/profile.controller.ts

```diff
@@ -0,0 +1,16 @@
+import Profile from '../entity/profile.entity';
+import { dataSource } from '../configs/dbConfig';
+import { Request, Response } from 'express';
+
+export const getProfile = async (req: Request, res: Response) => {
+  try {
+    const profileRepository = dataSource.getRepository(Profile);
+
+    // Find and return an array of all table rows in the profile table.
+    const allProfiles = await profileRepository.find();
+    res.status(200).json({ allProfiles });
```

Comment:
This should return only the logged-in user's profile. The request will contain the authorization header with the jwt token we are getting from the sign-in. That jwt token contains the userId. Using that you can fetch the profile.